tests/client/core/test_utils.js

```diff
@@ -207,3 +214,120 @@ describe('isValidClientApp', () => {
+  const addon = sinon.stub();
+  const state = {
+    addons: {
+      'the-addon': addon,
+    },
+  };
+
+  it('finds the add-on in the state', () => {
+    assert.strictEqual(findAddon(state, 'the-addon'), addon);
+  });
+
+  it('does not find the add-on in the state', () => {
+    assert.strictEqual(findAddon(state, 'different-addon'), undefined);
+  });
+});
+
+describe('loadAddonIfNeeded', () => {
+  const apiState = { token: 'my.jwt.token' };
+  const loadedSlug = 'my-addon';
+  let loadedAddon;
+  let dispatch;
+
+  beforeEach(() => {
+    loadedAddon = sinon.stub();
+    dispatch = sinon.spy();
+  });
+
+  function makeProps(slug) {
+    return {
+      store: {
+        getState: () => ({
+          addons: {
+            [loadedSlug]: loadedAddon,
+          },
+          api: apiState,
+        }),
+        dispatch,
+      },
+      params: { slug },
+    };
+  }
+
+  it('returns the add-on if loaded', () => {
+    assert.strictEqual(loadAddonIfNeeded(makeProps(loadedSlug)), loadedAddon);
+  });
+
+  it('loads the add-on if it is not loaded', () => {
+    const slug = 'other-addon';
+    const props = makeProps(slug, apiState);
+    const addon = sinon.stub();
+    const entities = { [slug]: addon };
+    const mockApi = sinon.mock(api);
+    mockApi
+      .expects('fetchAddon')
+      .once()
+      .withArgs({ slug, api: apiState })
+      .returns(Promise.resolve({ entities }));
+    const action = sinon.stub();
+    const mockActions = sinon.mock(actions);
+    mockActions
+      .expects('loadEntities')
+      .once()
+      .withArgs(entities)
+      .returns(action);
+    return loadAddonIfNeeded(props).then(() => {
+      assert(dispatch.calledWith(action), 'dispatch not called');
+      mockApi.verify();
+      mockActions.verify();
+    });
+  });
+
+  it('handles 404s when loading the add-on', () => {
+    const slug = 'other-addon';
+    const props = makeProps(slug, apiState);
+    const mockApi = sinon.mock(api);
+    mockApi
+      .expects('fetchAddon')
+      .once()
+      .withArgs({ slug, api: apiState })
+      .returns(Promise.reject(new Error('Error accessing API')));
+    const mockActions = sinon.mock(actions);
+    mockActions
+      .expects('loadEntities')
+      .never();
+    return loadAddonIfNeeded(props)
+      .then(unexpectedSuccess,
+        () => {
+          assert(!dispatch.called, 'dispatch called');
+          mockApi.verify();
+          mockActions.verify();
+        });
+  });
+});
+
+describe('nl2br', () => {
+  it('converts \n to <br/>', () => {
+    assert.equal(nl2br('\n'), '<br />');
+  });
+
+  it('converts \r to <br/>', () => {
+    assert.equal(nl2br('\r'), '<br />');
+  });
+
+  it('converts \r\n to <br/>', () => {
+    assert.equal(nl2br('\r\n'), '<br />');
+  });
+
+  it('converts \n\n to <br/>', () => {
+    assert.equal(nl2br('\n\n'), '<br /><br />');
+  });
+
+  it('converts \r\n\r\n to <br/>', () => {
+    assert.equal(nl2br('\r\n\r\n'), '<br /><br />');
```

Comment:
Should normalize multiple sets of line breaks to a single `<br />` like the description string suggests?